clang/include/clang/Basic/Attr.td

```diff
@@ -1607,6 +1607,13 @@ def SYCLDevice : InheritableAttr {
   let Documentation = [SYCLDeviceDocs];
 }
 
+def SYCLDeviceOnly : InheritableAttr {
+  let Spellings = [GNU<"sycl_device_only">];
```

Comment:
Is there a reason for not supporting the Clang spelling? Note that the Clang spelling implies GNU spelling.
```suggestion
  let Spellings = [Clang<"sycl_device_only">];
```